File: core/src/main/java/hello/core/MemberApp.java
package hello.core;

import hello.core.member.Grade;
import hello.core.member.Member;
import hello.core.member.MemberService;
import org.springframework.context.ApplicationContext;
import org.springframework.context.annotation.AnnotationConfigApplicationContext;

public class MemberApp {
    public static void main(String[] args) {

        // Vanilla Java Code
//        AppConfig appConfig = new AppConfig();
//        MemberService memberService = appConfig.memberService();

        // Spring Code
        // Bean 어노테이션으로 붙은 메소드를 DI 컨테이너에 넣고 관리하기 시작한다.
        ApplicationContext applicationContext = new AnnotationConfigApplicationContext(AppConfig.class);

        // 첫번째 인자에 Bean 어노테이션으로 등록한 메소드 명을 작성한다.
        MemberService memberService = applicationContext.getBean("memberService", MemberService.class);

        Member member = new Member(1L, "memberA", Grade.VIP);
        memberService.join(member);

        Member findMember = memberService.findMember(1L);

        System.out.println(findMember);
    }
}

File: src/main/kotlin/hello/core/MemberApp.kt
package hello.core

import hello.core.member.Grade
import hello.core.member.Member

class MemberApp {
}

fun main(args: Array<String>) {
    val appConfig = AppConfig()
    val memberService = appConfig.memberService()
    val member = Member(1L, "memberA", Grade.VIP)
    memberService.join(member)

    val findMember = memberService.findMember(1L)
    if (findMember != null) {
        println("findMember = ${findMember.name}")
    }

}